pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -53,11 +53,7 @@ func (configgen *ConfigGeneratorImpl) buildGatewayListeners(
 	if features.ScopeGatewayToNamespace.Get() {
 		gatewaysForWorkload = push.Gateways(node)
 	} else {
-		var workloadLabels labels.Collection
-		for _, w := range node.ServiceInstances {
-			workloadLabels = append(workloadLabels, w.Labels)
-		}
-		gatewaysForWorkload = env.Gateways(workloadLabels)
+		gatewaysForWorkload = env.Gateways(node.WorkloadLabels)
```

Comment:
is this always the same output?

Reply:
no, it is also determined by the gateway resources.